specs/yoga/clusterapi-driver.rst

```diff
@@ -0,0 +1,200 @@
+| Ingress controllers      | Octavia, Traefik,    | Nginx installed via Helm. |
+|                          | Nginx.               |                           |
++--------------------------+----------------------+---------------------------+
+| Horizon dashboard        | Supported.           | Not supported.            |
++--------------------------+----------------------+---------------------------+
+| Delegated authorisation  | Keystone trust.      | Application credential.   |
++--------------------------+----------------------+---------------------------+
+| Kubernetes CRD API       | None.                | Helm, flux, argo, etc.    |
++--------------------------+----------------------+---------------------------+
+| In-place upgrades        | Partial - depends on | Various supported         |
+|                          | driver.              | strategies, build in      |
+|                          |                      | infrastrucutre agnostic   |
+|                          |                      | code. Defaults to rolling |
+|                          |                      | upgrade (like Magnum).    |
++--------------------------+----------------------+---------------------------+
+| Self-healing             | Partial / uncertain. | Supported with            |
+|                          |                      | infrastrcuture agnostic   |
+|                          |                      | code via reconciliation   |
+|                          |                      | loop.                     |
++--------------------------+----------------------+---------------------------+
+| Auto-scaling             | Supported for        | Supported with            |
+|                          | default node group.  | infrastructure agnositc   |
+|                          |                      | code.                     |
++--------------------------+----------------------+---------------------------+
+| Multiple node groups     | Supported.           | Supported, with no default|
+|                          |                      | group.                    |
++--------------------------+----------------------+---------------------------+
+| Additional networks      | Supported (review    | Supported                 |
+|                          | pending)             |                           |
++--------------------------+----------------------+---------------------------+
+| New Kubernetes versions  | Test burden on Magnum| Test burden split between |
+|                          | entirely.            | Cluster API for           |
+|                          |                      | Kubernetes, Magnum for    |
+|                          |                      | infrastructure provision. |
++--------------------------+----------------------+---------------------------+
+
+The intention for the new driver is to make it compatible with the
+Magnum REST API.
```

Comment:
Maybe be explicit about we don't current expect and required changes to the REST API.